servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/SslCertificateMaxSizeTest.java

```diff
@@ -0,0 +1,129 @@
+    void respectsCertificateMaxSize(boolean serverLowerLimit, boolean clientLowerLimit) throws Exception {
+        final byte[] serverCertChain = readAllBytes(DefaultTestCerts.loadServerPem());
+        final byte[] clientCertChain = readAllBytes(DefaultTestCerts.loadClientPem());
+        // FIXME: To test the lower bound the cert length must be larger than BoringSSL's lower limit.
+        if (serverLowerLimit) {
+            assumeTrue(clientCertChain.length > BORINGSSL_LOWER_LIMIT);
+        }
+        if (clientLowerLimit) {
+            assumeTrue(serverCertChain.length > BORINGSSL_LOWER_LIMIT);
+        }
+        try (ServerContext server = serverBuilder(serverLowerLimit ? 1 : clientCertChain.length)
+                .listenBlockingAndAwait((ctx, request, responseFactory) ->
+                        responseFactory.ok().payloadBody("Hello World!", textSerializerUtf8()))) {
+            try (BlockingHttpClient client = clientBuilder(server, clientLowerLimit ? 1 : serverCertChain.length)
+                    .buildBlocking()) {
+                if (serverLowerLimit || clientLowerLimit) {
+                    assertThrows(SSLHandshakeException.class, () -> client.request(client.get("/sayHello")));
+                } else {
+                    assertThat(client.request(client.get("/sayHello")).status(), equalTo(OK));
+                }
+            }
+        }
+    }
+
+    private static HttpServerBuilder serverBuilder(int maxCertBytes) {
+        ServerSslConfigBuilder sslConfig = new ServerSslConfigBuilder(DefaultTestCerts::loadServerPem,
+                DefaultTestCerts::loadServerKey)
+                .trustManager(DefaultTestCerts::loadClientCAPem)
+                .clientAuthMode(REQUIRE)
+                .maxCertificateListBytes(maxCertBytes)
+                .provider(OPENSSL);
+        return HttpServers.forAddress(localAddress(0)).sslConfig(sslConfig.build());
+    }
+
+    private static SingleAddressHttpClientBuilder<HostAndPort, InetSocketAddress> clientBuilder(
+            ServerContext ctx, int maxCertBytes) {
+        ClientSslConfigBuilder sslConfig = new ClientSslConfigBuilder(DefaultTestCerts::loadServerCAPem)
+                .keyManager(DefaultTestCerts::loadClientPem, DefaultTestCerts::loadClientKey)
+                .peerHost(serverPemHostname())
+                .maxCertificateListBytes(maxCertBytes)
+                .provider(OPENSSL);
+        return HttpClients
+                .forSingleAddress(serverHostAndPort(ctx))
+                .sslConfig(sslConfig.build());
+    }
+
+    // JDK9+ can use InputStream.readAllBytes()
+    private static byte[] readAllBytes(InputStream is) throws IOException {
+        try {
+            final int readSize = 2048;
+            ByteArrayOutputStream buf = new ByteArrayOutputStream(readSize);
+            byte[] bytes = new byte[readSize];
+            while (is.read(bytes) > 0) {
+                buf.write(bytes);
+            }
```

Comment:
You may not read the full buffer on each iteration.
```suggestion
            byte[] bytes = new byte[readSize];
            int length;
            while ((length = is.read(bytes)) != -1) {
                buf.write(bytes, 0, length);
            }
```
Since it seems that you're only using this to get the length of the data stream you could skip the `ByteArrayOutputStream` stuff and just sum up the non `-1` values you get from the read calls and return that.